tests/modsys/T5.icry.stdout

```diff
@@ -1,7 +1,12 @@
 Loading module Cryptol
 Loading module Cryptol
-Loading module T5::A
-Loading module T5::B
-Loading module T5::Main
-main : [8]
-0x11
+Loading module T5::Main$argument
+
+Parse error at ./T5/B.cry:4:3--4:4
+  Module argument may not be a functor
```

Comment:
Is this test supposed to be failing? If it is:
1. I think the error message should be improved, it doesn't seem like "Module argument may not be a functor" is applicable here.
2. We should probably add a line to the docs stating that partially instantiating parameterized modules is no longer supported.

This same error happens with `T6.icry` and `T12.icry` (which have this error committed) and `tests/issues/issue796.icry` (which does not have it committed).

Reply:
This fails because anonymous argument module to `A` (i..e., the `where` block) is a functor, but it is not allowed to be.  Perhaps we should rephrase the error, please just change it if you have ideas, the message is in `ParserUtils` I think.

Partial module instantiations never worked, but this sort of pattern used to mean that you are defining a functor.  We decided that this notation is confusing (because the `parameters` are for `B` but they appear in the argument to `A`), so we should not support it.  I am not sure how official this feature ever was, but feel free to add a note to the docs to clarify that this does not work anymore.

Reply:
In 70d96006a5526a94dc72d5ee3bc84096770b7e3b I changed this message to "Instantiation of a parameterized module may not itself be parameterized" since this error is only raised when desugaring an anonymous instantiation of a parameterized module.